egs/demask/README.md

```diff
@@ -0,0 +1,219 @@
+signal processing tools with it !
+
+- Demask is trained on synthetic data generated from LibriSpeech's ([Panayotov et al. 2015](https://ieeexplore.ieee.org/document/7178964))
+clean speech, distorted by approximate surgical mask finite impulse response (FIR) filters taken from [Corey et al. 2020](https://arxiv.org/abs/2008.04521).
+The synthetic data is then augmented using room impulse responses (RIRs) from the FUSS dataset
+([Wisdom et al. 2020](https://zenodo.org/record/3743844#.X0JZEhl8Jkg)).
+A simple neural network estimates a time-frequency mask to correct the speech distortions.
+Thanks to Asteroid's filterbanks (formulated using `torch.nn`), we could use a time domain loss with a time-frequency model
+which yielded better results.
+
+## Asteroid highlights
+- Train a speech separation model in 20 lines.
+```python
+from torch import optim
+from pytorch_lightning import Trainer
+
+from asteroid import ConvTasNet
+from asteroid.losses import PITLossWrapper
+from asteroid.data import LibriMix
+from asteroid.engine import System
+
+train_loader, val_loader = LibriMix.loaders_from_mini(task='sep_clean', batch_size=4)
+model = ConvTasNet(n_src=2)
+optimizer = optim.Adam(model.parameters(), lr=1e-3)
+loss = PITLossWrapper(
+    lambda x, y: (x - y).pow(2).mean(-1),
+    pit_from="pw_pt",
+)
+
+
+system = System(model, optimizer, loss, train_loader, val_loader)
+trainer = Trainer(fast_dev_run=True)
+#trainer.fit(system)
+```
+
+
+- Load a pretrained model using Asteroid
+```python
+from asteroid.models import DPRNNTasNet
+
+model = DPRNNTasNet.from_pretrained("mpariente/DPRNNTasNet(ks=16)_WHAM!_sepclean")
+```
+
+- Load a pretrained model
+```python
+# Without installing Asteroid
+from torch import hub
+model = hub.load("mpariente/asteroid", "demask", "popcornell/DeMask_Surgical_mask_speech_enhancement_v1")
+
+# With asteroid install from master
+from asteroid import DeMask
+model = DeMask.from_pretrained("popcornell/DeMask_Surgical_mask_speech_enhancement_v1")
+```
+
+- Directly use it to enhance your muffled speech recordings using the `asteroid-infer` CLI
+```bash
+asteroid-infer popcornell/DeMask_Surgical_mask_speech_enhancement_v1 --files muffled.wav
+```
+
+To find the name of the model, we browsed our Zenodo community and picked the DeMask pretrained model
+
+TODO: fix the image is low  quality
+<p align="center">
+  <img src="https://iili.io/dUF0va.md.jpg" width="80%" />
+</p>
+
+## Challenges we ran into
+
+- In our first approach, we wanted to use the Compare dataset (classification with unpaired data),
+and use style transfer to perform enhancement but the amount of data was too small and the
+differences between mask vs. no-mask too subtle.
+When we got the impulse responses (IRs) from [Corey et al. 2020](https://arxiv.org/abs/2008.04521),
+none of our first ideas worked because the filters contained the IR of the microphone and the room, and the phase
+was noisy. Approximating the filters by hand to be linear-phase saved us !
```

Comment:
We then resorted to design ad-hoc FIR filters which directly approximate the frequency response of the masks in [Corey et al. 2020](https://arxiv.org/abs/2008.04521). These filters are dynamically generated at training time to augment the available data.